### PR TITLE
MAINT Add some more types to various things

### PR DESCRIPTION
**Pull request by hoodmane**

### CODE REVIEW

File: src/core/error_handling.ts [hunk 1]
```diff
@@ -74,14 +75,16 @@ let fatal_error_occurred = false;
  * @argument e {Error} The cause of the fatal error.
  * @private
  */
-API.fatal_error = function (e: any) {
+API.fatal_error = function (e: any): never {
   if (e && e.pyodide_fatal_error) {
+    // @ts-ignore
```
Comment (ryanking13) [inline src/core/error_handling.ts:80]: Why this function has `never` return type?

File: src/js/lib.ts
Comment (ryanking13) [inline src/js/lib.ts:1]: I am not sure if `lib` is the best name for this file. As it mostly defines types, maybe something like `types.ts` would be better?

Comment (hoodmane) [reply src/js/lib.ts:1]: > mostly

I would say entirely

File: src/core/error_handling.ts (hunk 1 above)
Comment (hoodmane) [reply src/core/error_handling.ts:80]: I want it to type narrow:
```ts
let x;
try {
  x = get number();
} catch(e) { 
    fatal_error(e);
}
// now I want x to have type number not type number | undefined
```

Comment (hoodmane) [reply src/core/error_handling.ts:80]: The exceptions to it not returning are in various cases that we're already in the process of fatally error teardown and we don't want to recursively fail. I could care less if there are more logic errors on this already dying code path and I don't want typescript trying to warn me about it.

Comment (ryanking13) [reply src/core/error_handling.ts:80]: Okay, I think it makes sense. Thanks for the explanation.

**Review by ryanking13 — APPROVED** [review]
Thanks, @hoodmane!

**Comment by hoodmane** [comment]
Thanks for the review @ryanking131

**Pull request merged**